### ISSUE TITLE
Memory leak in native code in app built with Jetpack Compose

### ISSUE THREAD

**Issue by Davidbrcz** [mapbox/mapbox-maps-android#2079]
## Environment

- Android OS version: 10, 11
- Devices affected:  One Plus 6 
- Maps SDK Version: at least  10.11.2'

## Observed behavior and steps to reproduce

The following application built with Jetpack compose has 2 navigation destinations, and with a button one can navigate between the 2. In one of them, a Mapbox map is created and shown through a AndroidView.

```kt
package com.example.mapboxleak

import android.content.Context
import android.os.Bundle
import android.view.Gravity
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material.Button
import androidx.compose.material.MaterialTheme
import androidx.compose.material.Surface
import androidx.compose.material.Text
import androidx.compose.runtime.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.viewinterop.AndroidView
import androidx.navigation.compose.NavHost
import androidx.navigation.compose.composable
import androidx.navigation.compose.rememberNavController
import com.example.mapboxleak.ui.theme.MapboxleakTheme
import com.mapbox.maps.*
import com.mapbox.maps.extension.observable.eventdata.MapLoadingErrorEventData
import com.mapbox.maps.extension.observable.model.MapLoadErrorType
import com.mapbox.maps.plugin.attribution.attribution
import com.mapbox.maps.plugin.delegates.listeners.OnMapLoadErrorListener
import com.mapbox.maps.plugin.gestures.gestures
import com.mapbox.maps.plugin.logo.logo
import com.mapbox.maps.plugin.scalebar.scalebar

object NavDest {
    const val H = "h"
    const val T = "T"
}

const val STYLE = Style.SATELLITE_STREETS

val EmptyStyle = """
    {
        "layers": [
            {
                "id": "water",
                "type": "background",
                "background-color": "#0000ff"
            }
        ],
       "sources": {},
        "version": 8
    }
""".trimIndent()

interface AttributionOptions {
    val gravity: Int
    val marginTop: Float
    val marginBottom: Float
    val marginStart: Float
    val marginEnd: Float

    companion object {
        val T = attributionOptions(Gravity.START or Gravity.BOTTOM, 0f, 400f, 0f, 0f)
    }
}

fun attributionOptions(
    gravity: Int,
    marginTop: Float,
    marginBottom: Float,
    marginStart: Float,
    marginEnd: Float,
): AttributionOptions = object : AttributionOptions {
    override val gravity: Int = gravity
    override val marginTop: Float = marginTop
    override val marginBottom: Float = marginBottom
    override val marginStart: Float = marginStart
    override val marginEnd: Float = marginEnd
}

@Composable
fun MapboxMap(
    modifier: Modifier = Modifier,
    enableGestures: Boolean = true,
    textureView: Boolean = true,
    attributionOptions: AttributionOptions = AttributionOptions.T,
    onStyleLoaded: ((MapView, MapboxMap, Style) -> Unit)? = null,
) {

    var theView by remember { mutableStateOf<MapView?>(null) }

    AndroidView(
        factory = { context ->
            theView?.let { return@AndroidView it }

            theView =
                MapView(context, enableGestures, attributionOptions, textureView).also { map ->

                    map.getMapboxMap().loadStyleUri(
                        styleUri = STYLE,
                        onStyleLoaded = {},
                        onMapLoadErrorListener = object : OnMapLoadErrorListener {
                            override fun onMapLoadError(eventData: MapLoadingErrorEventData) {
                                if (eventData.type == MapLoadErrorType.STYLE) {
                                    map.getMapboxMap().loadStyleJson(EmptyStyle)
                                }
                            }
                        },
                    )
                }

            theView!!
        },
        modifier = Modifier.then(modifier),
    )
}

private fun MapView(
    context: Context,
    gesturesEnabled: Boolean,
    attributionOptions: AttributionOptions,
    textureView: Boolean = true,
): MapView {

    val glyphsRasterizationOptions = GlyphsRasterizationOptions.Builder()
        .rasterizationMode(GlyphsRasterizationMode.ALL_GLYPHS_RASTERIZED_LOCALLY).build()
    val mapOptions = MapOptions.Builder().applyDefaultParams(context)
        .glyphsRasterizationOptions(glyphsRasterizationOptions).build()

    val theView = MapView(
        context,
        MapInitOptions(context, textureView = textureView, mapOptions = mapOptions),
    )

    theView.attribution.updateSettings {
        position = attributionOptions.gravity
        marginTop = attributionOptions.marginTop
        marginBottom = attributionOptions.marginBottom
        marginLeft = attributionOptions.marginStart
        marginRight = attributionOptions.marginEnd
    }
    theView.logo.updateSettings {
        position = attributionOptions.gravity
        marginTop = attributionOptions.marginTop
        marginBottom = attributionOptions.marginBottom
        marginLeft = attributionOptions.marginStart + 40
        marginRight = attributionOptions.marginEnd
    }
    theView.scalebar.updateSettings { enabled = false }
    theView.gestures.updateSettings {
        pitchEnabled = false
        scrollEnabled = gesturesEnabled
        rotateEnabled = false
        pinchToZoomEnabled = gesturesEnabled
        doubleTapToZoomInEnabled = gesturesEnabled
    }

    return theView
}

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            val navController = rememberNavController()
            MapboxleakTheme {
                // A surface container using the 'background' color from the theme
                Surface(
                    modifier = Modifier.fillMaxSize(),
                    color = MaterialTheme.colors.background,
                ) {
                    NavHost(
                        navController = navController,
                        startDestination = NavDest.H,
                    ) {
                        composable(NavDest.H) {
                            Column {
                                Button(onClick = {
                                    navController.navigate(NavDest.T)
                                }) {
                                    Text(text = "Go to test screen")
                                }
                            }
                          }
                        composable(NavDest.T) {
                            Column() {
                                Button(onClick = {
                                    navController.navigate(NavDest.H)
                                }) {
                                    Text(text = "Go HOME")
                                }
                                MapboxMap()
                            }
                        }
                    }
                }
            }
        }
    }
}


```

Each time we toggle, the native memory consumption goes up by ~12 MB. 

![leak-mapbox-cut](https://user-images.githubusercontent.com/5786937/234201351-81e6db95-4fb2-46bf-b010-810fa0a93105.png)

Native memory recording after toggling twice 
![native-alloc-leak](https://user-images.githubusercontent.com/5786937/234201358-054e7c47-20b1-43b5-82cf-0a03f4f50424.png)

## Expected behavior

Memory doesn't leak

## Additional links and references

Seems to be the same [SO](https://stackoverflow.com/questions/75920533/memory-leak-in-jetpack-compose-mapbox) issue

**Comment by kawan2019** [comment]
i have the same problem.  -  v10.12.2

**Comment by pengdev** [comment]
Hey @Davidbrcz have you tried our [compose extension](https://github.com/mapbox/mapbox-maps-android/tree/main/extension-compose) and see if you experience similar issues ?

**Comment by dryaz** [comment]
> Hey @Davidbrcz have you tried our [compose extension](https://github.com/mapbox/mapbox-maps-android/tree/main/extension-compose) and see if you experience similar issues ?

Here's the same issue with extensions and even in Mapboxs' own sample.

**Comment by pengdev** [comment]
Thanks @dryaz I will create an internal ticket and prioritise the fix.

**Comment by jush** [comment]
https://github.com/mapbox/mapbox-maps-android/releases/tag/v11.0.0-rc.2 contains the fix for #2174 which should also help with this issue.
